src/charm.py

```diff
@@ -167,7 +167,14 @@ def __init__(self, *args):
             port=8080,
         )
 
-        self.ingress_ssh = IngressPerUnitRequirer(self, relation_name="ingress-ssh", mode="tcp")
+        # if the unit is the leader we set the port as well.
```

Comment:
Explain this better, we only wants the leader to set the port because of blah blah.